bytecomp/emitcode.ml

```diff
@@ -50,24 +50,33 @@ let () =
     )
 
 (* Buffering of bytecode *)
-let out_buffer = ref(LongString.create 0)
+let create_bigarray = Bigarray.Array1.create Bigarray.Char Bigarray.c_layout
+
+let copy_bigarray src dst size =
+  for i = 0 to size - 1 do
+    let open Bigarray.Array1 in
+    set dst i (get src i)
+  done
+
+let out_buffer = ref(create_bigarray 0)
 and out_position = ref 0
 
 let extend_buffer needed =
-  let size = LongString.length !out_buffer in
+  let size = Bigarray.Array1.dim !out_buffer in
   let new_size = ref(max size 16) (* we need new_size > 0 *) in
   while needed >= !new_size do new_size := 2 * !new_size done;
-  let new_buffer = LongString.create !new_size in
-  LongString.blit !out_buffer 0 new_buffer 0 (LongString.length !out_buffer);
+  let new_buffer = create_bigarray !new_size in
+  copy_bigarray !out_buffer new_buffer size;
```

Comment:
This should be something like this (untested):
```ocaml
Bigarray.Array1.(blit !out_buffer (sub new_buffer 0 size));
```
(Bigarray.sub shares memory with the larger array, so you blit parts by blitting whole subarrays)